tfresource/lib.w

```diff
@@ -0,0 +1,101 @@
+bring "cdktf" as cdktf;
+bring fs;
+bring cloud;
+bring util;
+
+pub class TerraformResource {
+  commands: MutJson;
+
+  new() {
+    this.setupProvider();
+
+    this.commands = {};
+
+    class Shell extends cdktf.TerraformResource {
+      c: MutJson;
+
+      new(commands: MutJson) {
+        super(terraformResourceType: "shell_script");
+        this.c = commands;
+      }
+
+      pub synthesizeAttributes(): Json {
+        return {
+          lifecycle_commands: Json.deepCopy(this.c),
+        };
+      }
+    }
+
+    new Shell(this.commands);
+  }
+
+  pub onCreate(handler: inflight (): Json) {
+    let prog = this.toExecutable("create", handler);
+    this.commands.set("create", "node {prog}");
+  }
+
+  pub onRead(handler: inflight (Json): Json) {
+    let prog = this.toExecutable("read", handler);
+    this.commands.set("read", "node {prog}");
+  }
+
+  pub onUpdate(handler: inflight (Json): Json) {
+    let prog = this.toExecutable("update", handler);
+    this.commands.set("update", "node {prog}");
+  }
+
+  pub onDelete(handler: inflight (Json): Json?) {
+    let prog = this.toExecutable("delete", handler);
+    this.commands.set("delete", "node {prog}");
+  }
+
+  toExecutable(name: str, handler: inflight (Json): Json?): str {
+    let workdir = std.Node.of(this).app.workdir;
+    let code: str = unsafeCast(handler)?._toInflight();
```

Comment:
@Chriscbr I feel we need to expose the capability to convert a closure to JavaScript code somehow from our SDK.